provider/agent.go

```diff
@@ -77,10 +77,24 @@ func agentResource() *schema.Resource {
 			},
 			"startup_script": {
 				ForceNew:    true,
-				Description: "A script to run after the agent starts.",
+				Description: "A script to run after the agent starts. The script should exit when it is done to signal that the agent is ready to accept connections.",
 				Type:        schema.TypeString,
 				Optional:    true,
 			},
+			"startup_script_timeout": {
+				Type:         schema.TypeInt,
+				Default:      300,
+				ForceNew:     true,
+				Optional:     true,
+				Description:  "Time in seconds until the agent ready status is marked as timed out, this happens when the startup script has not completed (exited) in the given time.",
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+			"shutdown_script": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+				Description: "A script to run before the agent is stopped. The script should exit when it is done to signal that the workspace can be stopped.",
```

Comment:
Updated description here, I considered commenting it out or adding `(NOT YET IMPLEMENTED.)`, but we'll likely do this soon after we get the startup script changes in.